src/AutoRest.CSharp/Common/Input/Utf8JsonReaderExtensions.cs

```diff
@@ -70,6 +70,46 @@ public static bool TryReadString(this ref Utf8JsonReader reader, string property
             return true;
         }
 
+        public static bool TryReadEnumValue(this ref Utf8JsonReader reader, string propertyName, ref object? value)
+        {
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException();
+            }
+
+            if (reader.GetString() != propertyName)
+            {
+                return false;
+            }
+
+            reader.Read();
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.String:
+                    value = reader.GetString() ?? throw new JsonException();
+                    break;
+                case JsonTokenType.Number:
+                    if (reader.TryGetInt32(out int intValue))
+                    {
+                        value = intValue;
+                    }
+                    else if (reader.TryGetSingle(out float floatValue))
```

Comment:
If we don't support float now, shall we raise exception directly?

Reply:
CADL actually [supports float](https://microsoft.github.io/cadl/language-basics/enums#values).

I didn't include the full float support in this PR, because:
1. There is an issue in the sample of CADL doc. I've asked for clarification.
2. I want to confirm the way we handle fixed enum in the PR, so that it would be easier to add the float support in a separate PR.